dbms/src/Storages/DeltaMerge/Index/InvertedIndex/Reader.cpp

```diff
@@ -0,0 +1,248 @@
+    case TypeIndex::Int32:
+        return std::make_shared<InvertedIndexFileReader<Int32>>(path);
+    case TypeIndex::UInt64:
+        return std::make_shared<InvertedIndexFileReader<UInt64>>(path);
+    case TypeIndex::Int64:
+        return std::make_shared<InvertedIndexFileReader<Int64>>(path);
+    case TypeIndex::Date:
+        return std::make_shared<InvertedIndexFileReader<UInt16>>(path);
+    case TypeIndex::DateTime:
+        return std::make_shared<InvertedIndexFileReader<UInt32>>(path);
+    case TypeIndex::Enum8:
+        return std::make_shared<InvertedIndexFileReader<Int8>>(path);
+    case TypeIndex::Enum16:
+        return std::make_shared<InvertedIndexFileReader<Int16>>(path);
+    case TypeIndex::MyDate:
+    case TypeIndex::MyDateTime:
+    case TypeIndex::MyTimeStamp:
+        return std::make_shared<InvertedIndexFileReader<UInt64>>(path);
+    case TypeIndex::MyTime:
+        return std::make_shared<InvertedIndexFileReader<Int64>>(path);
+    default:
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported type_id: {}", magic_enum::enum_name(type_id));
+    }
+}
+
+InvertedIndexReaderPtr InvertedIndexReader::view(const DataTypePtr & type, ReadBuffer & buf, size_t index_size)
+{
+    auto type_id = type->isNullable() ? dynamic_cast<const DataTypeNullable &>(*type).getNestedType()->getTypeId()
+                                      : type->getTypeId();
+    switch (type_id)
+    {
+    case TypeIndex::UInt8:
+        return std::make_shared<InvertedIndexMemoryReader<UInt8>>(buf, index_size);
+    case TypeIndex::Int8:
+        return std::make_shared<InvertedIndexMemoryReader<Int8>>(buf, index_size);
+    case TypeIndex::UInt16:
+        return std::make_shared<InvertedIndexMemoryReader<UInt16>>(buf, index_size);
+    case TypeIndex::Int16:
+        return std::make_shared<InvertedIndexMemoryReader<Int16>>(buf, index_size);
+    case TypeIndex::UInt32:
+        return std::make_shared<InvertedIndexMemoryReader<UInt32>>(buf, index_size);
+    case TypeIndex::Int32:
+        return std::make_shared<InvertedIndexMemoryReader<Int32>>(buf, index_size);
+    case TypeIndex::UInt64:
+        return std::make_shared<InvertedIndexMemoryReader<UInt64>>(buf, index_size);
+    case TypeIndex::Int64:
+        return std::make_shared<InvertedIndexMemoryReader<Int64>>(buf, index_size);
+    case TypeIndex::Date:
+        return std::make_shared<InvertedIndexMemoryReader<UInt16>>(buf, index_size);
+    case TypeIndex::DateTime:
+        return std::make_shared<InvertedIndexMemoryReader<UInt32>>(buf, index_size);
+    case TypeIndex::Enum8:
+        return std::make_shared<InvertedIndexMemoryReader<Int8>>(buf, index_size);
+    case TypeIndex::Enum16:
+        return std::make_shared<InvertedIndexMemoryReader<Int16>>(buf, index_size);
+    case TypeIndex::MyDate:
+    case TypeIndex::MyDateTime:
+    case TypeIndex::MyTimeStamp:
+        return std::make_shared<InvertedIndexMemoryReader<UInt64>>(buf, index_size);
```

Comment:
Why do they decay to UInt64? Are there any references

Reply:
https://github.com/pingcap/tiflash/blob/3cda2f61923f47c89db026bd118eded90e76c625/dbms/src/TiDB/Schema/TiDBTypes.h#L26-L55

Reply:
https://github.com/pingcap/tiflash/blob/3cda2f61923f47c89db026bd118eded90e76c625/dbms/src/DataTypes/DataTypeMyTimeBase.h#L23